### ISSUE TITLE
Didn't work when using VS2019.

### ISSUE THREAD

**Issue by What-a-stupid-username** [nvpro-samples/optix_advanced_samples#10]
![Z2413T)AU8@9B$UFP$ {$3W](https://user-images.githubusercontent.com/37427780/69480434-57175a00-0e42-11ea-903f-34c2806e8724.png)
cmd.exe exit with error code 1.

I'm using VS2019 with platform toolset v141.

**Comment by droettger** [comment]
Please check if setting the CUDA_HOST_COMPILER inside the CMake GUI solves the issue:
Check this post: https://devtalk.nvidia.com/default/topic/1066449/optix/build-errors-in-optix-advanced-samples/post/5400874/#5400874

**Comment by bipul-mohanto** [comment]
Hi @droettger ! Is there any plan to update the code repository for latest OptiX 7.4 and CUDA 11.5, with MSVS 2019? Seems still it is not compatible (Dec 16, 2021).

**Comment by droettger** [comment]
There is no plan to update the OptiX 5 based examples inside this repository.

I've only adjusted them to compile and work with the latest available OptiX SDK 6.5.0 release still using the old OptiX API.
But since these examples are still written against OptiX 5, they are not making use of the ray-triangle hardware intersection functionality available on Turing and Ampere RTX boards which require to use the built-in triangle primitives added in OptiX 6, so these are considered legacy examples.

Instead I ported some of the introduction examples to the new OptiX 7 API and added some additional examples showing different multi-GPU rendering distribution and OpenGL interop techniques, and CUDA peer-to-peer resource sharing via NVLINK here: https://github.com/NVIDIA/OptiX_Apps
These work with OptiX 7.4.0 and MSVS 2019 and CUDA 11.5.

I skipped porting the simpler introduction examples because the OptiX SDK examples and this SIGGRAPH course on OptiX 7 cover the more basic things: https://github.com/ingowald/optix7course

You'll find all that information when looking into the sticky posts on the OptiX developer forum:
https://forums.developer.nvidia.com/c/gaming-and-visualization-technologies/visualization/optix/167
